docs/source/user_guides/basic_conceptions.rst

```diff
@@ -0,0 +1,302 @@
+        Generate response for fallback node, according to the previous node
+        we have been to.
+        If the previous node was `start_node`, a sample message will be returned,
+        otherwise the message will include user input.
+        """
+        if get_previous_node_name(ctx) == "start_node":
+            return Message(text="You should've started the dialog with 'Hello!'")
+        elif ctx.last_request is not None:
+            last_request = ctx.last_request.text
+            note = f"You should've written 'Ping', not '{last_request}'!"
+            return Message(text=f"That was against the rules! {note}")
+        else:
+            raise RuntimeError("Error occurred: last request is None!")
+            
+
+    ping_pong_script = {
+        "ping_pong_flow": {
+            "start_node": {
+                RESPONSE: Message(),
+                TRANSITIONS: {
+                    lbl.forward(): cnd.exact_match(Message(text="Hello!")),
+                },
+            },
+            "greeting_node": {
+                RESPONSE: Message(text="Hi! Let's play ping-pong!"),
+                TRANSITIONS: {
+                    lbl.forward(): cnd.regexp(r"^[P|p]ing!?$"),
+                },
+            },
+            "ping_pong_node": {
+                RESPONSE: Message(text="Pong!"),
+                TRANSITIONS: {
+                    lbl.repeat(): cnd.regexp(r"^[P|p]ing!?$"),
+                },
+            },
+            "fallback_node": {
+                RESPONSE: fallback_response,
+                TRANSITIONS: {
+                    "greeting_node": cnd.true(),
+                },
+            },
+        },
+    }
+
+    pipeline = Pipeline.from_script(
+        ping_pong_script,
+        start_label=("ping_pong_flow", "start_node"),
+        fallback_label=("ping_pong_flow", "fallback_node"),
+    )
+
+    if __name__ == "__main__":
+        pipeline.run()
+
+That's what we've changed:
+
+* ``fallback_node`` has a callback response, it prints different messages depending on the
+  previous node.
+
+.. note::
+
+    See `tutorial on response functions`_.
+
+* A special function ``get_previous_node_name`` was written to determine the name of the previous
+  visited node. It utilizes ``labels`` attribute of the ``Context`` object.
+
+.. note::
+
+    See `documentation of Context object`_.
+
+* Transitions were changed: transitions to next, previous and current node were replaced with special
+  standard transitions.
+
+.. note::
+
+    See `tutorial on transitions`_.
+
+* Conditions were changed: now regular expressions are used to check user text input value.
+
+.. note::
+
+    See `tutorial on conditions`_.
+
+Further exploration
+~~~~~~~~~~~~~~~~~~~
+
+There are still a lot of capabilities of Dialog Flow Framework that remain uncovered by this tutorial.
+
+For example:
+
+* You can use ``GLOBAL`` transitions that will be available from every node in your script.
+  See `tutorial on global transitions`_.
```

Comment:
Also, why are links in this section not in a `.. note::` container like links in other sections?

Reply:
That's what I thought about this:
- In sections before we were describing what we have already done and the links were not necessary for user to see to understand the changes. So they were wrapped in `note` container, meaning that they are something additional.
- In this section we are describing what else there is in DFF that we haven't covered, and user won't be able to understand that without seeing the linked tutorials. So the links are not something additional, they are essential part of the text.